lib/SPIRV/SPIRVReader.cpp

```diff
@@ -3400,7 +3400,15 @@ void SPIRVToLLVM::transIntelFPGADecorations(SPIRVValue *BV, Value *V) {
     SmallString<256> AnnotStr;
     generateIntelFPGAAnnotation(BV, AnnotStr);
     if (!AnnotStr.empty()) {
-      auto *GS = Builder.CreateGlobalStringPtr(AnnotStr);
+      Constant *GS = nullptr;
+      std::string StringAnnotStr = AnnotStr.c_str();
+      static unordered_map<std::string, Constant *> Annotations;
+      if (Annotations.find(StringAnnotStr) != Annotations.end())
+        GS = Annotations[StringAnnotStr];
+      else {
```

Comment:
Just to be consistent with the `else` block.
```suggestion
      if (Annotations.find(StringAnnotStr) != Annotations.end()) {
        GS = Annotations[StringAnnotStr];
      } else {
```